uber/templates/preregistration/stripeForm.html

```diff
@@ -1,16 +1,161 @@
+        $btn.prop('disabled', false);
+        if (result.error) {
+            toastr.error(result.error);
+        } else if (result.stripe_intent) {
+            var dollars = result.stripe_intent.amount / 100;
+            dollars = dollars.toLocaleString("en-US", {style:"currency", currency:"USD"});
+            var cancel_url = result.cancel_url || '../preregistration/cancel_payment'
+            $('#stripeModal').find('.modal-title').text('Paying ' + dollars + ' to {{ c.ORGANIZATION_NAME }}');
+            $('#stripeModal').find('.charge_desc').text(result.stripe_intent.description);
+            collectStripePayment(result.stripe_intent.client_secret, result.stripe_intent.id, result.success_url, cancel_url);
+        }
+        });
+};
+
+var collectStripePayment = function (client_secret, stripe_id, success_url, cancel_url) {
+    $('#stripeModal').modal('show');
+    toastr.clear();
+    var form = document.getElementById('payment-form');
+    var stripe = Stripe('{{ c.STRIPE_PUBLIC_KEY }}');
+    var elements = stripe.elements();
+    var card = elements.create("card");
+
+    $('#stripeModal').on('hidden.bs.modal', function (e) {
+        card.destroy();
+        $.post(cancel_url, {stripe_id: stripe_id, csrf_token: csrf_token},
+        function(result) {
+            if (result.message) {
+                toastr.error(result.message);
+                $(form).addClass('hidden');
+                $('#stripeModal').off('hidden.bs.modal');
+            }
+        });
+    })
+
+    // Set up Stripe.js and Elements to use in checkout form
+    var style = {
+        base: {
+            color: '#32325d',
+            fontFamily: '"Helvetica Neue", Helvetica, sans-serif',
+            fontSmoothing: 'antialiased',
+            fontSize: '16px',
+            '::placeholder': {
+            color: '#aab7c4'
+            }
+        },
+        invalid: {
+            color: '#fa755a',
+            iconColor: '#fa755a'
+        }
+    };
+
+    card.mount("#card-element");
+
+    card.addEventListener('change', ({error}) => {
+        const displayError = $('#card-errors');
+        if (error) {
+        displayError.textContent = error.message;
+        } else {
+        displayError.textContent = '';
+        }
+    });
+
+    $(form).removeClass('hidden');
+
+    form.addEventListener('submit', function(event) {
+        event.preventDefault();
+        stripe.confirmCardPayment(client_secret, {
+            payment_method: {
+                card: card
+            }
+            }).then(function(result) {
+            if (result.error) {
+                // Show error to your customer (e.g., insufficient funds)
+                toastr.error(result.error.message);
+            } else {
+                // The payment has been processed!
+                if (result.paymentIntent.status === 'succeeded') {
+                    if(success_url == '') {
+                        // Special override for at-door check-in modal payment workflow
```

Comment:
Our registration index page lets you open a "Check In" modal, which itself lets you* either mark an attendee as paid or take a Stripe payment. This means you open a modal from a modal, which is generally ill-advised. We use `loadCheckInForm()` to close one modal before opening the other.

The internet tells you that you can dynamically load a function name using `window[]`, so I tried to hook this up to use `success_url` as either a URL or a callable function, but that wasn't working so I just hardcoded the function name for now. _Hopefully_ we'll not need to call Stripe from a modal elsewhere anytime soon.